libclamav/regex_list.c

```diff
@@ -162,13 +161,35 @@ cl_error_t regex_list_match(struct regex_matcher *matcher, char *real_url, const
     struct cli_ac_data mdata;
     struct cli_ac_result *res = NULL;
 
-    assert(matcher);
-    assert(real_url);
-    assert(display_url);
+    if (NULL == matcher) {
```

Comment:
This would make an excellent macro in the style of the above.  E.g.,
``` c
#define VALIDATE_PARAM_NOT_NULL(retval_rcvr, param_name, done_label) \
    do { \
        if (NULL == param_name) { \
            retval_rcvr = CL_ENULLARG; \
            cli_errmsg("%s: " #param_name " must be initialized\n", __func__); \
            goto done_label; \
        } \
    } while (0)